### PR TITLE
Have a universal2 wheel be created for macos in addition to the standard x86_64 wheel.

### PR DESCRIPTION
**Pull request by dbaxa**
Signed-off-by: David Black <dblack@atlassian.com>

### CODE REVIEW

**Comment by dbaxa** [comment]
This is untested & I am making use of the azure runners to see what is generated.

**Comment by dbaxa** [comment]
Do not merge this - this is not working on m1 macs.

``` 
<snip>
File "/Users/user/.local/share/virtualenvs/my-venv-eXZNiRsb/lib/python3.10/site-packages/pyorc/__init__.py", line 13, in <module>
    from pyorc._pyorc import _orc_version
ImportError: dlopen(/Users/user/.local/share/virtualenvs/my-venv-eXZNiRsb/lib/python3.10/site-packages/pyorc/_pyorc.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace '__ZN3orc10ParseErrorD1Ev'
(my-venv) ```

**Comment by dbaxa** [comment]
This is now working :-) - so I'll clean this up.

**Comment by dbaxa** [comment]
This works as tested on a mac os 12 ec2 m1 instance -> 

```
brew install pipenv 
brew install python@3.10

# setup a pipenv for 3.10
# here ^^^^^

# download wheels from ci 
curl -output wheels.zip "https://noirello.visualstudio.com/d41b569e-7f1c-42e1-91a5-ea56fcfc7ea2/_apis/build/builds/485/artifacts?artifactName=wheels&api-version=7.0&%24format=zip"

unzip wheels.zip
pip install wheels/pyorc-0.7.0-cp310-cp310-macosx_10_9_universal2.whl

git clone https://github.com/noirello/pyorc.git
cd pyorc

CERT_PATH=$(python -m certifi)
export SSL_CERT_FILE=${CERT_PATH}
export REQUESTS_CA_BUNDLE=${CERT_PATH}

pip install pybind11

python setup.py build_ext --download-only
cd deps/orc-1.7.5
cmake -DCMAKE_BUILD_TYPE=DEBUG -DBUILD_JAVA=OFF  -DBUILD_LIBHDFSPP=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON
make -j 8 package 

cp -a examples ../
cp -a ./_CPack_Packages/Darwin/TGZ/ORC-1.7.5-Darwin/ ../
cd ../../

pip install -U pytest pytest-cov "pytest-xdist[psutil]" coverage codecov
pytest -vs --junitxml=./test-results.xml tests/ -n auto

```

which should have
```====================================================================== 261 passed in 73.19s (0:01:13) =======================================================================
``` as the output.

You can also install the python3.8 & 3.9 wheel files & test them in a similar fashion (e.g. `pipenv install --python /System/Volumes/Data/Library/Developer/CommandLineTools/usr/bin/python`, `pip install pybind11` & then `pip install wheels/pyorc-0.7.0-cp38-cp38-macosx_10_9_universal2.whl`. You do not need to re-build the `orc` dep and can run the tests in the `pyorc` repository after installing the required test deps)

File: azure-pipelines.yml [hunk 1]
```diff
@@ -203,6 +203,32 @@ stages:
...
+
+      - bash: |
+          set -o errexit
+          python3 -m pip install --upgrade pip
+          python3 -m pip install cibuildwheel==2.8.0
+        displayName: Install cibuildwheel dependencies
+      - bash: cibuildwheel --output-dir wheelhouse .
+        env:
+          ABI_VERSION: 'cp36'
+          CMAKE_OSX_ARCHITECTURES: 'x86_64;arm64'
```
Comment (dbaxa) [inline azure-pipelines.yml:223]: We need this - see https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_ARCHITECTURES.html for details.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -203,6 +203,32 @@ stages:
           pathtoPublish: wheelhouse
           artifactName: wheels
 
+  - job: macos_universal2
+    pool:
+      vmImage: 'macOS-12'
+    steps:
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: '3.10'
```
Comment (dbaxa) [inline azure-pipelines.yml:212]: I seemingly tried but failed to only have a python `3.10` wheel created. :shrug: .

**Comment by dbaxa** [comment]
Additionally, the `universal2` wheel files will also work on `amd64` macs (I tested using a similar setup as found in the above comment on an ec2 mac amd64 system).

**Comment by dbaxa** [comment]
@noirello can we get this merged & m1 wheel files made available :-) ?

File: azure-pipelines.yml (hunk 1 above)
Comment (dbaxa) [reply azure-pipelines.yml:223]: (Related discussion: https://github.com/pypa/cibuildwheel/pull/704).

Comment (noirello) [reply azure-pipelines.yml:223]: Why is the `ABI_VERSION` set to `cp36`? I couldn't find a reference for the environment variable in cibuildwheel or cmake.

**Review by noirello — CHANGES_REQUESTED** [review]
I had a few remarks, plus I merged #56, so this branch can be rebased on the master.

File: azure-pipelines.yml (hunk 1 above)
Comment (dbaxa) [reply azure-pipelines.yml:223]: AFAIK it isn't required, it is something I ended up copying from https://github.com/pyca/cryptography/blob/main/.github/workflows/wheel-builder.yml#L93 .

Comment (dbaxa) [reply azure-pipelines.yml:218]: Removed :heavy_check_mark:

**Review by noirello — APPROVED** [review]
Great, thank you. I'd like to look into a few things before making a new release (and uploading the universal wheels). I'll try to create the release until the end of the week.

**Pull request merged**